fbw-a380x/src/wasm/systems/a380_systems/src/fuel/fuel_quantity_management_system.rs

```diff
@@ -0,0 +1,614 @@
+
+    fn refuel_rate(&self) -> RefuelRate {
+        self.refuel_rate_setting
+    }
+
+    fn refuel_is_enabled(&self) -> bool {
+        self.refuel_status
+            && self.engine_1_state == EngineState::Off
+            && self.engine_2_state == EngineState::Off
+            && self.engine_3_state == EngineState::Off
+            && self.engine_4_state == EngineState::Off
+            && self.is_on_ground
+            && self.ground_speed < Velocity::new::<knot>(0.1)
+    }
+}
+impl SimulationElement for RefuelPanelInput {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.total_desired_fuel_input =
+            Mass::new::<kilogram>(reader.read(&self.total_desired_fuel_id));
+        self.refuel_status = reader.read(&self.refuel_status_id);
+        self.refuel_rate_setting = reader.read(&self.refuel_rate_setting_id);
+        self.ground_speed = reader.read(&self.ground_speed_id);
+        self.is_on_ground = reader.read(&self.is_on_ground_id);
+        self.engine_1_state = reader.read(&self.engine_1_state_id);
+        self.engine_2_state = reader.read(&self.engine_2_state_id);
+        self.engine_3_state = reader.read(&self.engine_3_state_id);
+        self.engine_4_state = reader.read(&self.engine_4_state_id);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(&self.refuel_status_id, self.refuel_status);
+        writer.write(
+            &self.total_desired_fuel_id,
+            self.total_desired_fuel_input.get::<kilogram>(),
+        );
+    }
+}
+
+pub struct IntegratedRefuelPanel {
+    powered_by: ElectricalBusType,
+    is_powered: bool,
+    _mode_select: ModeSelect,
+    input: RefuelPanelInput,
+}
+impl IntegratedRefuelPanel {
+    pub fn new(context: &mut InitContext, powered_by: ElectricalBusType) -> Self {
+        Self {
+            powered_by,
+            is_powered: false,
+            _mode_select: ModeSelect::AutoRefuel,
+            input: RefuelPanelInput::new(context),
+        }
+    }
+
+    fn total_desired_fuel(&self) -> Mass {
+        self.input.total_desired_fuel_input()
+    }
+
+    fn refuel_status(&self) -> bool {
+        self.input.refuel_status()
+    }
+
+    fn set_fuel_desired(&mut self, fuel: Mass) {
+        self.input.set_total_desired_fuel_input(fuel)
+    }
+
+    fn set_refuel_status(&mut self, status: bool) {
+        self.input.set_refuel_status(status);
+    }
+
+    fn refuel_rate(&self) -> RefuelRate {
+        self.input.refuel_rate()
+    }
+
+    fn refuel_is_enabled(&self) -> bool {
+        self.input.refuel_is_enabled()
+    }
+}
+impl SimulationElement for IntegratedRefuelPanel {
+    fn receive_power(&mut self, buses: &impl ElectricalBuses) {
+        self.is_powered = buses.is_powered(self.powered_by);
+    }
+
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.input.accept(visitor);
+        visitor.visit(self);
+    }
+}
+
+pub struct CoreProcessingInputsOutputsCommandModule {
```

Comment:
You should use the `CoreProcessingInputOutputModule` implementation to check whether your correspondent cpioms are active and to which other cpiom they can communicate. Only implement the "software" running in the cpiom here.

Reply:
Not sure what you mean here, need some clarification

Reply:
What I meant is that we already have the logic if the CPIOM is powered or not (see here: https://github.com/flybywiresim/aircraft/blob/master/fbw-common/src/wasm/systems/systems/src/integrated_modular_avionics/core_processing_input_output_module.rs). You can just use it like it's used for the air conditioning: https://github.com/flybywiresim/aircraft/blob/master/fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/cpiom_b.rs

Reply:
Ah ok I understand, well the CPIOM implementation that was planned in the end I never got that far with it, so I will just rename the struct so that when its refactored into CPIOM modules this can be done then